doc/python-interface/index.md

```diff
@@ -0,0 +1,11 @@
+(python_spglib)=
+
+# Spglib for Python
+
+```{toctree}
+---
+maxdepth: 1
+---
+Introduction <python-interface>
+API Reference <python-apidocs/spglib/spglib>
+```
```

Comment:
This document can be merged with `Introduction`/`python-interface.md`. Then add `self` in the `toctree`.